docs/testplan/Packet_Trimming_Testplan.md

```diff
@@ -0,0 +1,536 @@
+2. Create a buffer profile setting `packet_discard_action=trim` and apply buffer profile.
+3. Create egress queue congestion.
+   1. Create scheduler and apply it to the egress queue.  
+   `sonic-db-cli CONFIG_DB hset "SCHEDULER|SCHEDULER_BLOCK_DATA_PLANE" "type" DWRR "weight" 15 "pir" 1`  
+   `sonic-db-cli CONFIG_DB hset 'QUEUE|Ethernet0|0' scheduler SCHEDULER_BLOCK_DATA_PLANE`
+   2. Send packets to make the buffer full.
+4. Send packets and verify trimming function works well on Ethernet0.
+5. Update the buffer profile setting `packet_discard_action=drop` and apply buffer profile.
+6. Send packets and verify packets are dropped directly.
+7. Update the buffer profile setting `packet_discard_action=trim` and apply buffer profile.
+8. Send packets and verify trimming function works well on Ethernet0.
+9. Toggle admin state 10 times on Ethernet0.
+10. After each toggle, verify the buffer profile and buffer configuration are applied correctly on Ethernet0.
+11. After each toggle, send traffic and verify trimming function works well on Ethernet0.
+12. Update packet trimming with asymmetric DSCP.
+13. Toggle admin state 10 times on Ethernet0 and Ethernet4.
+14. After each toggle, verify the `TC_TO_DSCP_MAP` and `PORT_QOS_MAP` are applied correctly.
+
+---
+
+## Test Case 10: Stability During Feature State Toggles
+**Objective**: Ensure no crashes occur during repeated trimming config changes.  
+**Test Steps**:
+1. Configure packet trimming in global level and bind buffer profile.
+2. Send packets and verify trimming function works well.
+3. Update the `packet_discard_action` value (`trim` -> `drop` -> `trim`) in the buffer profile 10 times.
+4. Verify trimming function works well consistently.
+5. Verify no crashes/core dumps observed.
+
+---
+
+## Test Case 11: SRv6 Compatibility with Trimming
+**Objective**: Validate SRv6 functions when trimming is enabled/disabled.  
+**Test Steps**:  
+1. Configure packet trimming in global level and bind buffer profile.
+2. Config SRv6 as pipe mode.
+3. Verify the trimming function works well for SRv6 SHIFT packet.
+   - Verify the SRv6 SHIFT packet is trimmed, verify the trimmed packet size is correct.
+   - Verify the DSCP value in outer header is updated and DSCP value in inner header is not updated.
+   - Verify the `segment left` decreased in SRH header.
+4. Verify the trimming function works well for SRv6 DECAP packet.
+   - Verify the SRv6 DECAP packet is trimmed, verify the trimmed packet size is correct.
+   - Verify SRv6 packet decapsulation success.
+   - Verify the DSCP value is not updated in the packet.  
+  (The DSCP value of the outer header of the SRv6 packet is updated by trimming, and the DSCP value of the inner header is not updated. After decapsulation, the outer header is discarded, leaving only the inner packet. Therefore, the DSCP value of the inner header is not updated.)
+
+---
+
+## Test Case 12: Verify Trimming in Generic Config Updater Test
```

Comment:
Could you also add one test case for strict queue configured on the trim queue?

Reply:
@kperumalbfn  This is a new reuqirements, not only to udpate the test plan, we also need to plan effort to update the test case. We will take into internal to review how and when could we commit it, and besides this one, do you still have some other comments?  If not can we have merge this one, and update the new requirement later?